From 1e1ce61d967cc992b44c5206cc0cfe21cd44c7ff Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Fri, 12 Jan 2024 22:21:55 +0800
Subject: [PATCH 19/21] po: add patch to disable parallel execution

- `msgfilter' might fail while processing de.po to generate de_CH.po, if
  there are too many prarallel jobs.
- We observed this behaviour on our AMD EPYC 7H12 buildbot, with -j129.
- Cyan tested without this patch on an Alder Lake Intel i7 machine, even
  with `make -j` (as much as GNU Make can assign), no such failure
  occurred.
---
 bootstrap.conf                                |  3 ++-
 conf/Makefile.extra-dist                      |  1 +
 ...ile.in.in-disable-parallel-execution.patch | 27 +++++++++++++++++++
 3 files changed, 30 insertions(+), 1 deletion(-)
 create mode 100644 po/gettext-patches/0005-po-Makefile.in.in-disable-parallel-execution.patch

diff --git a/bootstrap.conf b/bootstrap.conf
index 7a7813d28..a7b3c657f 100644
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -90,7 +90,8 @@ bootstrap_post_import_hook () {
       0001-Support-POTFILES-shell \
       0002-Handle-gettext_printf-shell-function \
       0003-Make-msgfmt-output-in-little-endian \
-      0004-Use-SHELL-rather-than-bin-sh; do
+      0004-Use-SHELL-rather-than-bin-sh \
+      0005-po-Makefile.in.in-disable-parallel-execution; do
     patch -d po -p3 \
       < "po/gettext-patches/$patchname.patch"
   done
diff --git a/conf/Makefile.extra-dist b/conf/Makefile.extra-dist
index 5e7126f98..d21702f11 100644
--- a/conf/Makefile.extra-dist
+++ b/conf/Makefile.extra-dist
@@ -112,6 +112,7 @@ EXTRA_DIST += po/gettext-patches/0001-Support-POTFILES-shell.patch
 EXTRA_DIST += po/gettext-patches/0002-Handle-gettext_printf-shell-function.patch
 EXTRA_DIST += po/gettext-patches/0003-Make-msgfmt-output-in-little-endian.patch
 EXTRA_DIST += po/gettext-patches/0004-Use-SHELL-rather-than-bin-sh.patch
+EXTRA_DIST += po/gettext-patches/0005-po-Makefile.in.in-disable-parallel-execution.patch
 
 EXTRA_DIST += po/POTFILES-shell.in
 EXTRA_DIST += po/README
diff --git a/po/gettext-patches/0005-po-Makefile.in.in-disable-parallel-execution.patch b/po/gettext-patches/0005-po-Makefile.in.in-disable-parallel-execution.patch
new file mode 100644
index 000000000..0cef9fafe
--- /dev/null
+++ b/po/gettext-patches/0005-po-Makefile.in.in-disable-parallel-execution.patch
@@ -0,0 +1,27 @@
+From 8e8976fa14c4fe5e88cf9998893565635500d5b3 Mon Sep 17 00:00:00 2001
+From: Cyan <cyanoxygen@aosc.io>
+Date: Fri, 12 Jan 2024 22:19:25 +0800
+Subject: [PATCH] po/Makefile.in.in: disable parallel execution
+
+---
+ gettext-runtime/po/Makefile.in.in | 4 ++++
+ 1 file changed, 4 insertions(+)
+
+diff --git a/gettext-runtime/po/Makefile.in.in b/gettext-runtime/po/Makefile.in.in
+index 6b25f0d91..eb6e9d943 100644
+--- a/gettext-runtime/po/Makefile.in.in
++++ b/gettext-runtime/po/Makefile.in.in
+@@ -92,6 +92,10 @@ DISTFILESDEPS = $(DISTFILESDEPS_$(DIST_DEPENDS_ON_UPDATE_PO))
+ 
+ all: all-@USE_NLS@
+ 
++# FIXME: msgfilter might fail at some time if there are too much parallel
++# jobs. This is observed while building GNU GRUB with -j129.
++
++.NOTPARALLEL:
+ 
+ .SUFFIXES:
+ .SUFFIXES: .po .gmo .sed .sin .nop .po-create .po-update
+-- 
+2.39.1
+
-- 
2.44.0

